test/e2e/nsx_ipblocksinfo_test.go

```diff
@@ -0,0 +1,168 @@
+	})
+	require.NoError(t, err)
+	defer func() {
+		testData.nsxClient.VPCConnectivityProfilesClient.Delete(defaultOrg, defaultProject, defaultVPCProfile)
+	}()
+
+	// Create VPC with the profile above
+	vpcId := "ipblocks-info-test"
+	err = testData.nsxClient.VPCClient.Patch(defaultOrg, defaultProject, vpcId, model.Vpc{})
+	require.NoError(t, err)
+	vpcAttachmentId := "default"
+	err = testData.nsxClient.VpcAttachmentClient.Patch(defaultOrg, defaultProject, vpcId, vpcAttachmentId, model.VpcAttachment{
+		VpcConnectivityProfile: common.String(fmt.Sprintf("/orgs/%s/projects/%s/vpc-connectivity-profiles/%s", defaultOrg, defaultProject, vpcProfileName)),
+	})
+	require.NoError(t, err)
+	defer func() {
+		testData.nsxClient.VpcAttachmentClient.Delete(defaultOrg, defaultProject, vpcId, vpcAttachmentId)
+	}()
+
+	// Create VPCNetworkConfigurations
+	vpcConfigName := "vpc-config-ipblocks-info"
+	_, err = testData.crdClientset.CrdV1alpha1().VPCNetworkConfigurations().Create(context.TODO(), &v1alpha1.VPCNetworkConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: vpcConfigName,
+		},
+		Spec: v1alpha1.VPCNetworkConfigurationSpec{
+			NSXProject: fmt.Sprintf("/orgs/%s/projects/%s", defaultOrg, defaultProject),
+			VPC:        vpcId,
+		},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	privateTGWIPCIDRs, externalIPCIDRs := getDefaultIPBlocksCidrs(t)
+	defer func() {
+		// Delete VPCNetworkConfigurations and check
+		err = testData.crdClientset.CrdV1alpha1().VPCNetworkConfigurations().Delete(context.TODO(), vpcConfigName, metav1.DeleteOptions{})
+		require.NoError(t, err)
+		assertIPBlocksInfo(t, privateTGWIPCIDRs, externalIPCIDRs)
+	}()
+
+	// Check IPBlocksInfo
+	updatedPrivateTGWIPCIDRs := append(privateTGWIPCIDRs, ipBlockCidr)
+	assertIPBlocksInfo(t, updatedPrivateTGWIPCIDRs, externalIPCIDRs)
+}
+
+func getDefaultIPBlocksCidrs(t *testing.T) (privateTGWIPCIDRs []string, externalIPCIDRs []string) {
+	vpcProfile, err := testData.nsxClient.VPCConnectivityProfilesClient.Get(defaultOrg, defaultProject, defaultVPCProfile)
+	require.NoError(t, err)
+	// Assume only one ipblock in default VPC Connectivity Profile
+	externalBlock := vpcProfile.ExternalIpBlocks[0]
+	privateTGWBlock := vpcProfile.PrivateTgwIpBlocks[0]
+
+	results, err := testData.queryResource(common.ResourceTypeIPBlock, []string{})
+	require.NoError(t, err)
+	res := transSearchResponsetoIPBlock(results)
+	count := 0
+	for _, ipblock := range res {
+		if count >= 2 {
+			break
+		}
+		if *ipblock.Path == externalBlock {
+			externalIPCIDRs = append(externalIPCIDRs, *ipblock.Cidr)
+			count++
+		}
+		if *ipblock.Path == privateTGWBlock {
+			privateTGWIPCIDRs = append(privateTGWIPCIDRs, *ipblock.Cidr)
+			count++
+		}
+	}
+	return
+}
+
+func assertIPBlocksInfo(t *testing.T, privateTGWIPCIDRs []string, externalIPCIDRs []string) {
+	deadlineCtx, deadlineCancel := context.WithTimeout(context.Background(), defaultTimeout)
```

Comment:
the sync period of ipblockinfo is 600 second. is it possible that ipblockinfo not finishing sync?

Reply:
It should be fine as 
- for InitialIPBlocksInfo case, the first sync starts at the 30th second
- for CustomIPBlocksInfo, the ipblocksinfo will be updated by UpdateIPBlocksInfo when VPCNetworkConfiguration creation event is received